controllers/followersController.js

```diff
@@ -0,0 +1,121 @@
+import createError from 'http-errors';
+import models from '../models';
+import Response from '../helpers/responseHelper';
+
+/**
+ * Class handling followers operation
+ *
+ * @class FollowersController
+ */
+class FollowersController {
+  /**
+   * User followers handler
+   *
+   * @static
+   * @param {object} request - Express Request object
+   * @param {object} response - Express Response object
+   * @returns {object} Response object
+   * @param {Function} next - Express NextFunction
+   * @memberof FollowersController
+   */
+  static async follow(request, response, next) {
+    const { params: { username } } = request;
+
+    try {
+      const { followable, follower } = await FollowersController.validateFollowable(request);
+      await followable.addFollower(follower);
+
+      return Response.send(response, 200, [], `Successfully followed ${username}`);
+    } catch (error) {
+      next(error);
+    }
+  }
+
+  /**
+   * User unfollow handler
+   *
+   * @static
+   * @param {object} request - Express Request object
+   * @param {object} response - Express Response object
+   * @returns {object} Response object
+   * @param {Function} next - Express NextFunction
+   * @memberof FollowersController
+   */
+  static async unfollow(request, response, next) {
+    const { params: { username } } = request;
+
+    try {
+      const { followable, follower } = await FollowersController.validateFollowable(request);
+      const existingFollower = await followable.hasFollowers(follower);
+      if (!existingFollower) {
+        next(createError(400, 'Sorry, you cannot unfollow a user you are not following'));
+      }
+
+      await followable.removeFollower(follower);
+      return Response.send(response, 200, [], `Successfully unfollowed ${username}`);
+    } catch (error) {
+      next(error);
+    }
+  }
+
+  /**
+   * Validate users to follow
+   *
+   * @static
+   * @param {object} request - Express Request object
+   * @returns {object} Object holding the information of the followable and follower
+   * @memberof FollowersController
+   */
+  static async validateFollowable(request) {
+    const { user, params: { username } } = request;
+
+    try {
+      const follower = await models.User.findOne({
```

Comment:
#### Critical
- Also, check if the user is not deactivated. ie `deletedAt === null`

Reply:
O yes, that's a good one. Will have that checked also. Thanks